.github/workflows/kvssink.yml

```diff
@@ -275,3 +275,107 @@ jobs:
+            automake
+            build-essential
+            cmake
+            git
+            gstreamer1.0-plugins-base-apps
+            gstreamer1.0-plugins-bad
+            gstreamer1.0-plugins-good
+            gstreamer1.0-plugins-ugly
+            gstreamer1.0-tools
+            libcurl4-openssl-dev
+            libgstreamer1.0-dev
+            libgstreamer-plugins-base1.0-dev
+            liblog4cplus-dev
+            libssl-dev
+            pkg-config
+            mkvtoolnix
+          use-cache: 'false' # Cache entry for Ubuntu 22.04 and 24.04 are the same (conflict)
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Move repository in WSL
+        run: |
+          # Copy to ~/kvs-cpp-repo for simplicity
+          # Note: Can't move due to no permissions
+          REPO_NAME=$(basename ${{ github.repository }})
+          cp -r /mnt/d/a/$REPO_NAME/$REPO_NAME ~/kvs-cpp-repo
+
+      - name: Build kvssink in WSL
+        run: |
+          mkdir -p ~/kvs-cpp-repo/build
+          cd ~/kvs-cpp-repo/build
+          cmake .. -DBUILD_GSTREAMER_PLUGIN=ON -DALIGNED_MEMORY_MODEL=ON -DBUILD_DEPENDENCIES=OFF
+          make -j$(nproc)
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          role-session-name: ${{ secrets.AWS_ROLE_SESSION_NAME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-duration-seconds: 900
+
+      - name: Run kvssink with dump dir in WSL
+        run: |
+          mkdir -p ~/kvs-cpp-repo/build/debug_output
+          export GST_PLUGIN_PATH=~/kvs-cpp-repo/build
+          export KVS_DEBUG_DUMP_DATA_FILE_DIR=~/kvs-cpp-repo/build/debug_output
+
+          set +e  # Disable exit on error for the timeout command
+          # Note: `env.` syntax to reference credentials to have GitHub Actions inject
+          # the secret values into these environment variables before the command runs,
+          # since the WSL shell doesn't have access to GitHub's environment variables
+          timeout --preserve-status --signal=SIGINT --kill-after=15s 30s \
+            gst-launch-1.0 videotestsrc is-live=true num-buffers=450 \
+              ! video/x-raw,framerate=30/1,width=640,height=480 \
+              ! videoconvert ! x264enc tune=zerolatency key-int-max=45 \
+              ! h264parse \
+              ! kvssink stream-name='demo-stream-cpp-kvssink-ci-wsl-videotestsrc-${{ matrix.image }}' \
+                  aws-region=${{ env.AWS_DEFAULT_REGION }} \
+                  access-key=${{ env.AWS_ACCESS_KEY_ID }} \
+                  secret-key=${{ env.AWS_SECRET_ACCESS_KEY }} \
+                  session-token=${{ env.AWS_SESSION_TOKEN }}
+          EXIT_CODE=$?
+          set -e  # Re-enable exit on error
+
+          if [[ $EXIT_CODE -ne 0 ]]; then
+            echo "gst-launch-1.0 exited with status: $EXIT_CODE"
```

Comment:
Might we want to also exit with "1" here to fail the test in this case?

Reply:
Yes